From 12f9faab961a007bfec98ee4751e19777a2e7b1d Mon Sep 17 00:00:00 2001
From: Larisa Grigore <larisa.grigore@nxp.com>
Date: Fri, 5 Feb 2021 17:12:06 +0200
Subject: [PATCH 62/78] serial:linflex: Disable DMA in linflex_flush_buffer

DMATXE should be disabled each time a dma operation is
terminated.

Issue: ALB-6257
Signed-off-by: Larisa Grigore <larisa.grigore@nxp.com>
---
 drivers/tty/serial/fsl_linflexuart.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/tty/serial/fsl_linflexuart.c b/drivers/tty/serial/fsl_linflexuart.c
index 0d2c376766f5..e9dd3e53a40d 100644
--- a/drivers/tty/serial/fsl_linflexuart.c
+++ b/drivers/tty/serial/fsl_linflexuart.c
@@ -468,6 +468,7 @@ static void linflex_flush_buffer(struct uart_port *port)
 					struct linflex_port, port);
 
 	if (sport->dma_tx_use) {
+		linflex_disable_dma_tx(port);
 		dmaengine_terminate_all(sport->dma_tx_chan);
 		sport->dma_tx_in_progress = 0;
 	}
-- 
2.25.1

